### PR TITLE
Allow classes without the proper interface to still work on JDK < 16

### PR DESCRIPTION
**Pull request by AndyCoveo**
To facilitate migration for people transitionning from the version 1 to version 2, the mechanism to set the detail message reflectively has been added back as a fallback mechanism. **This fallback mechanism is only available while running on JDK < 16**.

### CODE REVIEW

File: src/test/java/com/coveo/feign/BaseServiceException.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package com.coveo.feign;
+
+public abstract class BaseServiceException extends Exception {
```
Comment (fullbej) [inline src/test/java/com/coveo/feign/BaseServiceException.java:3]: why did you add this exception type ? It would have worked with current ServiceException no ? just curious

**Review by fullbej — APPROVED** [review]

File: src/main/java/com/coveo/feign/ReflectionErrorDecoder.java [hunk 2]
```diff
@@ -129,6 +127,14 @@ public Exception decode(String methodKey, Response response) {
   }
 
   private void initialize() {
+    try {
+      detailMessageField = Throwable.class.getDeclaredField("detailMessage");
+      detailMessageField.setAccessible(true);
+    } catch (Exception e) {
+      logger.debug("Unable to directly set the detailMessage, make sure exception do implement {}", baseExceptionClass.getName());
```
[the remark this answers is not in the document]
Comment (fullbej) [reply src/main/java/com/coveo/feign/ReflectionErrorDecoder.java:134]: nvm we just discussed the fact that we now have 2 messages, the warning is still here

File: src/main/java/com/coveo/feign/ReflectionErrorDecoder.java [hunk 3]
```diff
@@ -2,10 +2,7 @@
 
 import java.io.IOException;
 import java.lang.annotation.Annotation;
-import java.lang.reflect.Constructor;
-import java.lang.reflect.InvocationTargetException;
-import java.lang.reflect.Method;
-import java.lang.reflect.Modifier;
+import java.lang.reflect.*;
```
Comment (mcnoreau12) [inline src/main/java/com/coveo/feign/ReflectionErrorDecoder.java:5]: Question de préférence mais on préfère les singles import plutôt que `*`. :D Tu peux re-switcher ça :plz:

**Review by mcnoreau12 — APPROVED** [review]
thanks! gg
